dex/pair/src/contexts/ctx_helper.rs

```diff
@@ -0,0 +1,272 @@
+        context.set_second_token_reserve(self.pair_reserve(&second_token_id).get());
+    }
+
+    fn load_lp_token_supply(&self, context: &mut dyn Context<Self::Api>) {
+        context.set_lp_token_supply(self.lp_token_supply().get());
+    }
+
+    fn load_initial_k(&self, context: &mut dyn Context<Self::Api>) {
+        let k = self.calculate_k_constant(
+            context.get_first_token_reserve(),
+            context.get_second_token_reserve(),
+        );
+        context.set_initial_k(k);
+    }
+
+    fn construct_add_liquidity_output_payments(
+        &self,
+        context: &mut AddLiquidityContext<Self::Api>,
+    ) {
+        let mut payments: ManagedVec<EsdtTokenPayment<Self::Api>> = ManagedVec::new();
+
+        payments.push(self.create_payment(
+            context.get_lp_token_id(),
+            0,
+            context.get_liquidity_added(),
+        ));
+        payments.push(self.create_payment(
+            context.get_first_token_id(),
+            0,
+            &(&context.get_first_payment().amount - context.get_first_amount_optimal()),
+        ));
+        payments.push(self.create_payment(
+            context.get_second_token_id(),
+            0,
+            &(&context.get_second_payment().amount - context.get_second_amount_optimal()),
+        ));
+
+        context.set_output_payments(payments);
+    }
+
+    fn construct_remove_liquidity_output_payments(
+        &self,
+        context: &mut RemoveLiquidityContext<Self::Api>,
+    ) {
+        let mut payments: ManagedVec<EsdtTokenPayment<Self::Api>> = ManagedVec::new();
+
+        payments.push(self.create_payment(
+            context.get_first_token_id(),
+            0,
+            context.get_first_token_amount_removed(),
+        ));
+        payments.push(self.create_payment(
+            context.get_second_token_id(),
+            0,
+            context.get_second_token_amount_removed(),
+        ));
+
+        context.set_output_payments(payments);
+    }
+
+    fn construct_swap_output_payments(&self, context: &mut SwapContext<Self::Api>) {
+        let mut payments: ManagedVec<EsdtTokenPayment<Self::Api>> = ManagedVec::new();
+
+        payments.push(self.create_payment(
+            context.get_token_out(),
+            0,
+            context.get_final_output_amount(),
+        ));
+
+        if context.get_final_input_amount() != context.get_amount_in() {
+            payments.push(self.create_payment(
+                context.get_second_token_id(),
+                0,
+                &(context.get_amount_in() - context.get_final_input_amount()),
+            ));
+        }
+
+        context.set_output_payments(payments);
+    }
+
+    fn execute_output_payments(&self, context: &dyn Context<Self::Api>) {
+        self.send_multiple_tokens_if_not_zero(
+            context.get_caller(),
+            context.get_output_payments(),
+            context.get_opt_accept_funds_func(),
+        )
+        .unwrap();
```

Comment:
Consider using the `assert!` macros instead of `unwrap()`. I think a signal error directly is more efficient than the `panic!` code of `unwrap()`.

Reply:
done